cmd/kubeadm/app/phases/upgrade/postupgrade.go

```diff
@@ -103,6 +106,39 @@ func PerformPostUpgradeTasks(client clientset.Interface, cfg *kubeadmapi.InitCon
 		errs = append(errs, err)
 	}
 
+	if err := PerformAddonsUpgrade(client, cfg, out); err != nil {
+		errs = append(errs, err)
+	}
+
+	return errorsutil.NewAggregate(errs)
+}
+
+// PerformAddonsUpgrade performs the upgrade of the coredns and kube-proxy addons
+func PerformAddonsUpgrade(client clientset.Interface, cfg *kubeadmapi.InitConfiguration, out io.Writer) error {
+	unupgradedControlPlanes, err := unupgradedControlPlaneInstances(client, cfg.NodeRegistration.Name)
+	if err != nil {
+		err = errors.Wrapf(err, "failed to determine whether all the control plane instances have been upgraded")
+		if features.Enabled(cfg.FeatureGates, features.UpgradeAddonsAfterControlPlane) {
+			return err
+		}
+
+		// when UpgradeAddonsAfterControlPlane feature gate is disabled, just throw a warning
+		klog.Warning(err)
+	}
+	if len(unupgradedControlPlanes) > 0 {
+		if features.Enabled(cfg.FeatureGates, features.UpgradeAddonsAfterControlPlane) {
+			fmt.Fprintf(out, "[upgrade/addons] skip upgrade addons because control plane instances %v have not been upgraded\n", unupgradedControlPlanes)
+			return nil
+		}
+
+		// when UpgradeAddonsAfterControlPlane feature gate is disabled, just throw a warning
+		klog.Warningf("upgrading addons when control plane instances %v have not been upgraded,"+
```

Comment:
wondering we should make this a `klog.V(1).Infof()`

users tend to watch for warnings and complain / log issues when something is confusing to them.
with a V1 / V2 we could hide this information during Alpha, WDYT?